language/evm/move-to-yul/src/generator.rs

```diff
@@ -193,8 +218,29 @@ impl Generator {
                 "mstore(${MEM_SIZE_LOC}, memoryguard(${USED_MEM}))"
             );
             self.need_move_function(&fun_id);
+
+            for (idx, arg) in args.iter().enumerate() {
+                emit!(ctx.writer, "let arg{} := ", idx);
```

Comment:
It would be great if you guys could double check if I'm using the right naming scheme here for the bindings of the arguments. I used `a0, a1, ...` initially and somehow the compiler complained about conflicting names.

Reply:
Internal names should be prefixed with `$`. Source level names will be used for locals where they are available (so if you have `a0` in the source, it would potentially conflict). This kind of to make generated code better readable for debugging.

However,  `$t[0-9]+` must be avoided because the stackless bytecode uses it already for temporaries... This is not completely clean, but not that bad that refactoring was worth the effort until now.